palace.pyx

```diff
@@ -1158,6 +1160,97 @@ cdef class SourceGroup:
         """
         self.impl.destroy()
 
+
+cdef class AuxiliaryEffectSlot:
+    """An effect processor.
+
+    It takes the output mix of zero or more sources,
+    applies DSP for the desired effect (as configured
+    by a given Effect object), then adds to the output mix.
+
+    Parameters
+    ----------
+    context : Optional[Context]
+        The context from which the auxiliary effect slot is to be created.
+        If it is `None`, the object is left uninitialized.
+    """
+    cdef alure.AuxiliaryEffectSlot impl
+
+    def __init__(self, context: Optional[Context]) -> None:
+        if context is None: return
+        self.impl = (<Context> context).impl.create_auxiliary_effect_slot()
```

Comment:
No `None`-checking or casting will be needed if type is `Context`.

Reply:
Gentle ping.

Reply:
I've already fixed this but it's reverted due to git mess 😫

Reply:
Wait, it's already fixed ([palace.pyx:1236](https://github.com/McSinyx/palace/blob/aux-eff/palace.pyx#L1236)), idk why this is not marked as outdated though

Reply:
I mean the `(<Context> context)` cast part.  That's only needed when it's not obvious for the compiler that `context` is of type `Context` (ooh I love Python naming convention).

Reply:
After you fix this, you can merge it yourself, but remember to use the **Squash and merge** in the drop down menu.  Cheers!